src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.cs

```diff
@@ -117,7 +117,7 @@ internal static void ValidateTypeParameter<T>()
         {
             Type type = typeof(T);
             if (type != typeof(byte)   && type != typeof(short) && type != typeof(int) && type != typeof(long) &&
-                type != typeof(double) && type != typeof(float) && type != typeof(decimal))
+                type != typeof(double) && type != typeof(float) && type != typeof(decimal) && type != typeof(uint))
```

Comment:
Should this also enable `ulong` and other types so that the full set of signed/unsigned types is supported?

Reply:
I think, I can change this method with below

```
if ( ! type.IsPrimitive || type == typeof(IntPtr) || type == typeof(UIntPtr)
```

Did you agree with me?

https://docs.microsoft.com/en-us/dotnet/api/system.type.isprimitive?view=net-6.0

Reply:
@jkotas https://stackoverflow.com/a/32337906/1539100

Reply:
Would allowing `ulong` hurt OpenTelemetry compatibility? [NumberDataPoint allows double or sfixed64](https://github.com/open-telemetry/opentelemetry-proto/blob/8672494217bfc858e2a82a4e8c623d4a5530473a/opentelemetry/proto/metrics/v1/metrics.proto#L408-L413), and `ulong` might not fit in either of those without losing accuracy. I'm not sure these measurements are forwarded as NumberDataPoint, though.

Reply:
@KalleOlaviNiemitalo  @jkotas  So for this time, I only add uint.